flask_project/campaign_manager/templates/campaign_detail.html

```diff
@@ -548,6 +550,36 @@ <h3 class="detail-campaign-name">{{ name }}</h3>
             map.addControl(switchMap)
         }
 
+        // get user's location here
+        function getLocation(){
+            var query_string = 'https://freegeoip.net/json/' + '{{ ip }}';
+            var latitude;
+            var longitude;
+            //console.log('{{ ip }}'); 
+            //console.log(query_string);
+
+            function plotmarker(){
+                var xhttp = new XMLHttpRequest();
+                xhttp.onreadystatechange = function(){
+                    if(this.readyState == 4 && this.status == 200){
+                        var response = JSON.parse(this.response);
+                        latitude = response.latitude;
+                        longitude = response.longitude;
+                        var mapMarkers = new L.FeatureGroup();
+                        map.addLayer(mapMarkers);
+                    
+                        center = {'lat':latitude, 'lng':longitude};
+                        var marker = L.marker(center, {icon: orangeIcon}).addTo(mapMarkers);
+                    } 
+                }
+                xhttp.open("GET", query_string, true);
```

Comment:
when I hit [the query string (https://freegeoip.net/json/83.192.91.235)](https://freegeoip.net/json/83.192.91.235):
```
{
  "0": "#################################################################################################################################",
  "1": "#                                                                                                                               #",
  "2": "# IMPORTANT - PLEASE UPDATE YOUR API ENDPOINT                                                                                   #",
  "3": "#                                                                                                                               #",
  "4": "# This API endpoint is deprecated and has now been shut down. To keep using the freegeoip API, please update your integration   #",
  "5": "# to use the new ipstack API endpoint, designed as a simple drop-in replacement.                                                #",
  "6": "# You will be required to create an account at https://ipstack.com and obtain an API access key.                                #",
  "7": "#                                                                                                                               #",
  "8": "# For more information on how to upgrade please visit our Github Tutorial at: https://github.com/apilayer/freegeoip#readme      #",
  "9": "#                                                                                                                               #",
  "a": "#################################################################################################################################"
}
```

Reply:
![capture d ecran 2018-07-10 a 17 38 30](https://user-images.githubusercontent.com/35932320/42521049-1d3958f2-8468-11e8-9595-367beaf8672b.png)